### DISCUSSION TITLE
npm TLS1.0 deprecation

### DISCUSSION THREAD

**Discussion by RSM-SBA** [appium/appium#15827] [Q&A] [answered]
I know thats not really an appium topic, but on one of our jenkins nodes i get this warnings when i only install appium with drivers and plugins via npm install. 

package.json
`"dependencies": {
    "appium": "^2.0.0-beta.13"
  },
  "scripts": {
    "postinstall": "npx appium driver install xcuitest && npx appium plugin install images"
  },`

log:
`+ npm install
npm notice Beginning October 4, 2021, all connections to the npm registry - including for package installation - must use TLS 1.2 or higher. You are currently using plaintext http to connect. Please visit the GitHub blog for more information: https://github.blog/2021-08-23-npm-registry-deprecating-tls-1-0-tls-1-1/`

I read the blog entry and made the test successfully.
My question is, does appium use http connections to download the packages ? and is that a problem than ?

`npm install -g https://tls-test.npmjs.com/tls-test-1.0.0.tgz

> tls-test@1.0.0 postinstall /usr/local/lib/node_modules/tls-test
> node ./index.js

Hello! The tls-test package was successfully downloaded and installed.

Congratulations! Your package manager appears to support TLS 1.2.`

**Comment by jlipps** [comment, marked as answer]
Appium just runs `npm` under the hood without specifying anything specific about TLS. Maybe you need to upgrade the npm client on the jenkins machine.
